packages/core/strapi/lib/Strapi.js

```diff
@@ -570,10 +571,36 @@ class Strapi {
   }
 }
 
-module.exports = (options) => {
-  const strapi = new Strapi(options);
-  global.strapi = strapi;
-  return strapi;
+let singleton;
+
+const initializer = (options) => {
+  singleton = new Strapi(options);
+  global.strapi = singleton;
```

Comment:
would it be worth wrapping the get of `global.strapi` to warn users this is deprecated and will be removed in the next iteration?

Reply:
Yes we need to migrate the code before it starts warning users. I was thinking adding an env var to toggle it on to help us migrate though. wdyt ?